### DISCUSSION TITLE
response cache plugin download count

### DISCUSSION THREAD

**Discussion by xddq** [dotansimha/envelop#1445] [General]
First of all, great job on envelop. Its pretty easy to use and get going!

I am just wondering, why caching plugins like  `@envelop/validation-cache` and `@envelop/parser-cache` average **80k** weekly downloads based on npmjs.

While the (in my opinion most helpful) plugin  `@envelop/response-cache` or `@envelop/response-cache-redis` only average **2k and 0.4k** weekly downloads respectively?

Is it that most people write the cache implementation themselves? Is this not a good solution when using envelop anyway? Should I use something different for caching responses server side? I am pretty clueless.

**Comment by n1ru4l** [comment]
> I am just wondering, why caching plugins like @envelop/validation-cache and @envelop/parser-cache average 80k weekly downloads based on npmjs.

These packages have higher download counts as they are part of [GraphQL Yoga (our batteries included GraphQL server)](https://www.graphql-yoga.com/). For most people, we recommend using Envelop through GraphQL Yoga instead of as a standalone. However, if you already have an existing GraphQL server and cannot migrate to GraphQL Yoga you can use envelop.

[We are also currently planning a dedicated Response Cache Plugin for Yoga that does further optimizations.](https://github.com/dotansimha/graphql-yoga/pull/1359)

> Is it that most people write the cache implementation themselves?

I assume most people don't use a response cache and instead cache stuff within their business logic in resolvers.

> Is this not a good solution when using envelop anyway?

There is no simple answer. It always depends. If you have a high traffic service and pretty sporadic cache invalidations adding a response cache can give you a huge performance boost. We heard great feedback from Redwood.js users that utilize the envelop response cache with envelop.

If you don't mind about vendor lock-in you can also have a look at https://stellate.co/, which is an edge proxy response-cache solution.

**Reply by xddq** [reply]
Interesting since graphql-yoga averages 15 downloads instead of the 80k.

I guess the business logic caching might just be the reason. Thanks : )

Maybe someone that uses caching for parsing and validation but not for execution could answer here.
